src/pages/account-ownership-transfer/[guid].tsx

```diff
@@ -0,0 +1,170 @@
+import * as React from 'react'
+import LoginRequired from 'components/login-required'
+import {useRouter} from 'next/router'
+import toast from 'react-hot-toast'
+import axios from 'axios'
+import {
+  getAuthorizationHeader,
+  getTokenFromCookieHeaders,
+  AUTH_DOMAIN,
+} from 'utils/auth'
+
+async function confirmAccountOwnershipTransfer(guid: string) {
+  let transferSucceeded
+
+  try {
+    await axios.patch(
+      `${process.env.NEXT_PUBLIC_AUTH_DOMAIN}/api/v1/account_ownership_transfer_invitations/${guid}/accept_invite`,
+      {},
+      {
+        headers: {...getAuthorizationHeader()},
+      },
+    )
+
+    transferSucceeded = true
+  } catch (e) {
+    transferSucceeded = false
+  }
+
+  return transferSucceeded
+}
+
+type accountOwnershipTransferData = {
+  ownerEmail?: string
+  validInvite: boolean
+}
+
+export async function getServerSideProps(context: any) {
+  const {guid} = context.params
+  const {eggheadToken} = getTokenFromCookieHeaders(
+    context.req.headers.cookie as string,
+  )
+
+  let props: accountOwnershipTransferData = {
+    validInvite: false,
+  }
+
+  try {
+    const {data}: {data: {owner_email: string}} = await axios.get(
+      `${AUTH_DOMAIN}/api/v1/account_ownership_transfer_invitations/${guid}`,
+      {
+        headers: {
+          Authorization: `Bearer ${eggheadToken}`,
+        },
+      },
+    )
+
+    props = {
+      validInvite: true,
+      ownerEmail: data.owner_email,
+    }
+  } catch (e) {
+    if (e.response.status === 404) {
+      props.validInvite = false
+    }
+  }
+
+  return {
+    props,
+  }
+}
```

Comment:
style: this was confusing for me because there is a mix and match style in this function. `let` is used so we can assign new references to `props` but then we also use property accessors to mutate the initial reference.

I think that **any** error would indicate an invalid invite?

This is how I'd think about it removing the variable entirely.

```suggestion
export async function getServerSideProps(context: any) {
  const {guid} = context.params
  const {eggheadToken} = getTokenFromCookieHeaders(
    context.req.headers.cookie as string,
  )

  try {
    const {data}: {data: {owner_email: string}} = await axios.get(
      `${AUTH_DOMAIN}/api/v1/account_ownership_transfer_invitations/${guid}`,
      {
        headers: {
          Authorization: `Bearer ${eggheadToken}`,
        },
      },
    )
    
    return {props: {
      validInvite: true,
      ownerEmail: data.owner_email,
    }}
  } catch (e) {
    res.setHeader('Location', initialLesson.path)
    res.statusCode = e.response.status
    console.error(e.response.data.msg, guid)
    res.end()
    return {props: {validInvite = false}}
  }
}
```